src/internal/document_api/mod.rs

```diff
@@ -435,53 +500,177 @@ pub fn encrypt_document<
     grant_to_author: bool,
     user_grants: &'a Vec<UserId>,
     group_grants: &'a Vec<GroupId>,
-    policy_grant: Option<&PolicyGrant>,
+    policy_grant: Option<&'a PolicyGrant>,
 ) -> impl Future<Item = DocumentEncryptResult, Error = IronOxideErr> + 'a {
     let (dek, doc_sym_key) = transform::generate_new_doc_key(recrypt);
```

Comment:
A comment here might be nice - I assume `dek` is the `Plaintext` and `doc_sym_key` is the sha256 hash of it?

Reply:
I would argue this is clear from the docs on `generate_new_doc_key`
```rust
/// Generate a DEK and it's associated symmetric key for a new document
pub fn generate_new_doc_key<CR: rand::CryptoRng + rand::RngCore>(
    recrypt: &Recrypt<Sha256, Ed25519, RandomBytes<CR>>,
) -> (Plaintext, DerivedSymmetricKey) {
```